test/net-hostlist-test.js

```diff
@@ -664,37 +778,51 @@ describe('Net HostList', function() {
       assert.strictEqual(hosts.needsFlush, false);
     }
 
-    // update refCount.
+    // update refCount. (we only have 1 refCount, increase up to 8)
     {
-      const srcs = [
-        NetAddress.fromHostname(getRandomIPv4(), regtest),
-        NetAddress.fromHostname(getRandomIPv4(), regtest)
-      ];
+      const srcs = [];
+      for (let i = 0; i < 7; i++)
+        srcs.push(getRandomNetAddr());
+
+      const factors = [];
+      const _random = hosts.random;
+      const random = function (factor) {
+        factors.push(factor);
+        return 0;
+      };
+      hosts.random = random;
+
       const addr2 = cloneAddr(addr);
 
-      // when we have 1 ref, so probability should be 50%.
-      // then we have 2 refs, probability will be 25%.
-      // ... until we have 8 refs.
+      // when we have 1 ref, so probability of adding second one
+      // is 50% (1/2).
+      // then we have 2 refs, probability of adding will be 25% (1/4).
+      // ... until we have 8 refs. (last one being 1/128)
```

Comment:
Second comment is incorrect, will fix.

As for probability, you before this comment, there's custom random setup which collects the factors calculated by the `add` method, which in turn is checked after adding is done.

Reply:
Ok I think I understand now - but also - because you stubbed `random()` none of this probability stuff actually happens? The addr is added 100% of the time, in this test.